### PR TITLE
Add support for opening sqlite db with shared cache

### PR DESCRIPTION
**Pull request by othercorey**
Refs https://github.com/cakephp/migrations/issues/513

### CODE REVIEW

File: src/Database/Driver/Sqlite.php [hunk 1]
```diff
@@ -132,12 +134,27 @@ public function connect(): bool
...
+        }
 
-        $dsn = "sqlite:{$config['database']}";
-        $this->_connect($dsn, $config);
+        $params = [];
+        if ($config['cache']) {
+            $params[] = 'cache=' . $config['cache'];
+        }
+        if ($config['mode']) {
+            $params[] = 'mode=' . $config['mode'];
```
Comment (markstory) [inline src/Database/Driver/Sqlite.php:148]: What does mode do?

Comment (othercorey) [reply src/Database/Driver/Sqlite.php:148]: https://www.sqlite.org/uri.html Section 3.3

There is a `memory` mode that let's you name the db but act like `:memory:` so you can create multiple in-memory dbs.

**Comment by othercorey** [comment]
I tried creating a `params` config, but it's very difficult to configure an array of key/value pairs through DSN format.

**Comment by markstory** [comment]
> I tried creating a params config, but it's very difficult to configure an array of key/value pairs through DSN format.

I think the unprefixed keys are generally a better design as it is easier to remember and operate.

**Pull request merged**